### ISSUE TITLE
Strings still in English even in de_DE (informal and formal)

### ISSUE THREAD

**Issue by Braintelligence** [vendidero/one-stop-shop-woocommerce#7]
Hello everyone,

I'm running this plugin on a few hours old (so relatively fresh) Wordpress installation. For some reason everything in `/wp-admin/admin.php?page=oss-reports` is in English, even though the website runs on `de_DE_formal`. 

Excluding the subtitle of `/wp-admin/admin.php?page=wc-settings&tab=germanized-oss` everything in there is also in English.

`/wp-admin/admin.php?page=wc-settings&tab=email&section=oss_woocommerce_deliverythresholdemailnotification` is also completely in English.

I don't understand this; the website is set to `de_DE_formal` and Loco Translate shows that a translation exists. WooCommerce and everything else is clearly in German, but almost everything about One Stop Shop is in English.

Any idea what could go wrong here? System report attached.

```
### WordPress Environment ###

WordPress address (URL): redacted
Site address (URL): redacted
WC Version: 7.1.0
REST API Version: ✔ 7.1.0
WC Blocks Version: ✔ 8.7.5
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.1.1
WP Multisite: –
WP Memory Limit: 2 GB
WP Debug Mode: –
WP Cron: ✔
Language: de_DE_formal
External object cache: ✔

### Server Environment ###

Server Info: Apache/2.4.38 (Debian)
PHP Version: 7.4.8
PHP Post Max Size: 2 GB
PHP Time Limit: 600
PHP Max Input Vars: 1000
cURL Version: 7.64.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 8.0.20
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Ihr Server verfügt nicht über eine lauffähige Version der SoapClient Klasse - einige Zahlungsschnittstellen
welche SOAP nutzen
funktionieren unter Umständen nicht korrekt.

DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.1.0
WC Database Prefix: wp_
Datenbank-Gesamtgröße: 6.05MB
Datenbank-Datengröße: 3.68MB
Datenbank-Indexgröße: 2.37MB
wp_woocommerce_sessions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Daten: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Daten: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_fbv: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_fbv_attachment_folder: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ninja_table_items: Daten: 0.16MB + Index: 0.00MB + Engine InnoDB
wp_options: Daten: 1.47MB + Index: 0.11MB + Engine InnoDB
wp_postmeta: Daten: 0.19MB + Index: 0.03MB + Engine InnoDB
wp_posts: Daten: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_social_users: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Daten: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Daten: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Daten: 0.09MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfhoover: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wfls_2fa_secrets: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wftrafficrates: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_gzd_dhl_im_product_services: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_dhl_im_products: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_packaging: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_gzd_packagingmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipment_itemmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipment_items: Daten: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gzd_shipment_labelmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipment_labels: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipmentmeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gzd_shipments: Daten: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_gzd_shipping_provider: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_gzd_shipping_providermeta: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wpfm_backup: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpforms_tasks_meta: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Daten: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yith_wcwl: Daten: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yith_wcwl_lists: Daten: 0.02MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 4
customize_changeset: 10
ninja-table: 34
page: 14
post: 3
revision: 2
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (20) ###

Contact Form 7: von Takayuki Miyoshi – 5.6.4
FileBird Lite: von Ninja Team – 5.1
OMGF: von Daan from Daan.dev – 5.5.0
Loco Translate: von Tim Whitlock – 2.6.3
Nextend Social Login: von Nextendweb – 3.1.7
Ninja Tables: von WPManageNinja LLC – 4.3.2
One Stop Shop für WooCommerce: von vendidero – 1.3.3
Redis Object Cache: von Till Krüss – 2.2.2
SVG Support: von Benbodhi – 2.5.4
Advanced Editor Tools (vormals TinyMCE Advanced): von Automattic – 5.6.0
Two Factor: von Mitwirkende – 0.7.3
UpdraftPlus - Backup/Restore: von UpdraftPlus.Com
DavidAnderson – 1.22.23

Min Max Quantity & Step Control for WooCommerce: von CodeAstrology Team – 3.4
Germanized für WooCommerce: von vendidero – 3.11.2
WooCommerce: von Automattic – 7.1.0
Wordfence Security: von Wordfence – 7.8.0
WP File Manager: von mndpsingh287 – 7.1.7
WP Font Awesome: von Zayed Baloch – 1.7.8
WP Mail SMTP: von WPForms – 3.6.1
YITH WooCommerce Wishlist: von YITH – 3.15.0

### Inactive Plugins (3) ###

Akismet Anti-Spam: von Automattic – 5.0.2
Checkout Field Editor for WooCommerce: von ThemeHigh – 1.8.0
Hello Dolly: von Matt Mullenweg – 1.7.2

### Dropin Plugins (1) ###

object-cache.php: Redis Object Cache Drop-In

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop-Basis: #8 - /shop/
Warenkorb: #9 - /cart/
Kasse: #10 - /checkout/
Mein Konto: #11 - /my-account/
Allgemeine Geschäftsbedingungen: #28 - /agb/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.16.2
Parent Theme Author URL: https://uxthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome-child/woocommerce/emails/admin-new-order.php Version 2.5.0 ist veraltet. Die Hauptversion ist 3.7.0
flatsome-child/woocommerce/emails/customer-on-hold-order.php Version 2.5.0 ist veraltet. Die Hauptversion ist 3.7.0
flatsome-child/woocommerce/emails/customer-processing-order.php Version 2.5.0 ist veraltet. Die Hauptversion ist 3.7.0
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php
flatsome/woocommerce/content-product_cat.php

Outdated Templates: ❌
					
					
						Erfahren Sie
wie Sie aktualisieren können


### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-12-08 13:22:41 +01:00
Options: ✔
Notes: 37
Onboarding: -

### Action Scheduler ###

Abgeschlossen: 9
Oldest: 2022-12-07 14:38:41 +0100
Newest: 2022-12-07 15:38:26 +0100

Ausstehend: 5
Oldest: 2022-12-08 03:00:00 +0100
Newest: 2022-12-12 14:00:00 +0100


### Status report information ###

Generated at: 2022-12-07 15:48:13 +01:00
```

**Comment by Braintelligence** [comment]
Huh, after looking at the changelog of 1.3.3 I suspected a regression.

I was right; with 1.3.2 everything works as expected.

1.3.3 must have introduced a regression for translations somewhere.

EDIT: Sadly Germanized doesn't recognize 1.3.2 as a proper installation... The name of the plugin changed.

**Comment by dennisnissle** [comment]
Hi there,

seems strange to me. We indeed added some extra-checking to support language variants (de_AT, de_CH). See: https://github.com/vendidero/one-stop-shop-woocommerce/blob/main/src/Package.php#L515

But there is very little room for bugs here as far as I can tell. We use the same logic in Germanized too. Is Germanized translated to German just fine?

**Comment by Braintelligence** [comment]
Hi,

Germanized settings in Woocommerce look all German to me.

EDIT: Reading this bit here `oss-woocommerce' === $domain` I'm asking myself if this would fail if the domain of the wordpress instance changed at some point.

I started the installation with `sub.domain.tld` but then changed it to `otherdomain.tld` via wp-config.php and changing two rows in the wp_options table.

**Comment by dennisnissle** [comment]
Could you try commenting out (or temporarily removing) [this](https://github.com/vendidero/one-stop-shop-woocommerce/blob/main/src/Package.php#L499) line in `one-stop-shop-woocommerce/src/Package.php`? Does hat fix the issue? 

The actual domain has nothing to do with the code. We are comparing the text domain (of the plugin).

**Comment by Braintelligence** [comment]
Activating the current version, then commenting out that line didn't help.

**Comment by dennisnissle** [comment]
That is strange as this is the only thing added between the two versions: https://github.com/vendidero/one-stop-shop-woocommerce/compare/v1.3.2...v1.3.3

Did you try to deactivate Germanized (to make sure that the OSS package from Germanized is not loaded first) and then do the code changes?

**Comment by Braintelligence** [comment]
Ok this is weird; if I deactivate Germanized the name of the plugin changes.
While Germanized is active it's called `Germanized für WooCommerce: One Stop Shop`. If it's deactivated it's called `One Stop Shop für WooCommerce`.
Other than that: It doesn't matter in what order I activate the plugins; OSS stays English while Germanized stays German.

Even if Germanized is deactivated OSS stays English.

**Comment by dennisnissle** [comment]
> While Germanized is active it's called Germanized für WooCommerce: One Stop Shop. If it's deactivated it's called One Stop Shop für WooCommerce.

This was chosen on purpose as the plugin is linked to Germanized.

> Even if Germanized is deactivated OSS stays English.

I was not able to reproduce that issue with the latest version (using de_DE or de_DE_formal as language) in a local setup. Could you please try to deactivate all plugins but Woo and OSS and activate a default theme? Does it work? 

Did you manually override the language file via `wp-content/languages/`?

**Comment by Braintelligence** [comment]
I deactivated all addons except for WooCommerce and OSS, didn't help.
Also changing the theme didn't help.
I emptied all of `wp-content/plugins/languages`, didn't help. :/

**Comment by dennisnissle** [comment]
That is very strange! Did you try reinstalling the whole plugin, e.g. deleting, downloading a fresh copy, new install? What is your current configured language in WP-Admin > Settings > General? And what is your current admin profiles' language?

**Comment by Braintelligence** [comment]
Current language is `de_DE_formal` aka `Deutsch (Sie)`.

One thing that I find weird also is the fact that the Germanized plugin has a German description in the plugin install (from web) dialogue, while one stop shop has an English dialogue.
I find this weird because when I visit the plugin website directly everything is in German.

**Comment by Braintelligence** [comment]
If it helps I could make a snapshot and give you access to the site. I'd just need a mail address.

**Comment by dennisnissle** [comment]
Could you contact us via https://vendidero.de/kontakt - thanks!

**Comment by dennisnissle** [comment]
Fixed in 1.3.4 - thanks for reporting!
